src/OrchardCore.Modules/OrchardCore.Notifications/Drivers/NotifyUserTaskDisplayDriver.cs

```diff
@@ -17,4 +25,33 @@ public class NotifyUserTaskDisplayDriver : NotifyUserTaskActivityDisplayDriver<N
         : base(htmlSanitizerService, liquidTemplateManager, notificationOptions, stringLocalizer)
     {
     }
+
+    protected override string EditShapeType { get; } = $"{ActivityName}_Fields_Edit";
+
+    public override async Task<IDisplayResult> UpdateAsync(NotifyUserTask activity, IUpdateModel updater)
+    {
+        var viewModel = new NotifyUserTaskViewModel();
+        await updater.TryUpdateModelAsync(viewModel, Prefix);
+
+        var userNames = viewModel.UserNames?.Split(',', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries)
+            .Distinct(StringComparer.OrdinalIgnoreCase) ?? [];
+
+        if (!userNames.Any())
+        {
+            updater.ModelState.AddModelError(Prefix, nameof(viewModel.UserNames), S["Please provide at least one username to notify."]);
+        }
+        else
+        {
+            activity.UserNames = new WorkflowExpression<string>(string.Join(',', userNames));
```

Comment:
For readability in the text box:

```suggestion
            activity.UserNames = new WorkflowExpression<string>(string.Join(", ", userNames));
```

Reply:
done